ord_schema/validations.py

```diff
@@ -410,12 +412,17 @@ def validate_reaction(message: reaction_pb2.Reaction,
                       options: Optional[ValidationOptions] = None):
     if options is None:
         options = ValidationOptions()
-    if len(message.inputs) == 0:
-        warnings.warn('Reactions should have at least 1 reaction input',
-                      ValidationError)
-    if len(message.outcomes) == 0:
-        warnings.warn('Reactions should have at least 1 reaction outcome',
-                      ValidationError)
+    if (options.allow_reaction_smiles_only and
+            message_helpers.get_reaction_smiles(message) and
```

Comment:
I think we should try to parse the reactants, any agents, and product(s) into RDKit and issue ValidationWarnings if it doesn't parse. Like we do for SMILES in check_compound_identifiers?

Reply:
Right now this branch only executes if there are no reactants/agents or products at all. I'm thinking that I should add a check for valid reaction SMILES and leave this as-is? If there are reactants or products they will be checked in the normal way.

Reply:
Yes, good idea